tests/profiling/collector/test_stack_asyncio.py

```diff
@@ -1,42 +1,69 @@
 import asyncio
 import collections
 import sys
+import types
 
 import pytest
 
 from ddtrace.profiling import _asyncio
 from ddtrace.profiling import profiler
 from ddtrace.profiling.collector import stack_event
+from ddtrace.profiling.collector.stack import StackCollector
 
 from . import _asyncio_compat
 
 
+def patch_stack_collector(stack_collector):
+    """
+    Patch a stack collect so we can count how many times it has run
+    """
+
+    def _collect(self):
+        self.run_count += 1
+        return self._orig_collect()
+
+    stack_collector.run_count = 0
+    orig = stack_collector.collect
+    stack_collector._orig_collect = orig
+    stack_collector.collect = types.MethodType(_collect, stack_collector)
```

Comment:
```python
stack_collector.collect = _collect.__get__(stack_collector)
```

Reply:
We also have utilities in `ddtrace.internal.wrapping` for wrapping/unwrapping functions/methods of already instantiated stuff. This could perhaps be made into a fixture with setup and teardown steps.